### PR TITLE
Hotfix: fixes CI by correcting cpp-logger and DLIO profiler/dftracer stuff

### PR DESCRIPTION
**Pull request by ilumsden**
While looking at CI for other PRs, I found that GitHub Actions was failing to build DYAD because it couldn't find cpp-logger. After looking into the issue, I realized that our check for the `<PackageName>_FOUND` variable after `find_package` was simply using the wrong variable name. It was using `CPP_LOGGER_FOUND`, while it should have been using `cpp-logger_FOUND`. This PR corrects that variable name.

After fixing this issue with cpp-logger, I also discovered that CI was failing due to @hariharan-devarajan's recent renaming of DLIO profiler to dftracer. This PR also modifies the Python bindings to use the new name (and other minor API changes in dftracer).

### CODE REVIEW

File: pydyad/setup.cfg [hunk 1]
```diff
@@ -8,4 +8,5 @@ classifier =
 python_requires = >=3.7
 install_requires =
     numpy
-    dlio_profiler_py @ git+https://github.com/hariharan-devarajan/dlio-profiler.git
\ No newline at end of file
+    # dlio_profiler_py @ git+https://github.com/hariharan-devarajan/dlio-profiler.git
+    pydftracer @ git+https://github.com/hariharan-devarajan/dftracer.git
```
Comment (hariharan-devarajan) [inline pydyad/setup.cfg:12]: Do u want to switch it to pydftrace==1.0.2. Its better than a moving target.

File: CMakeLists.txt [hunk 2]
```diff
@@ -231,7 +231,7 @@ if (DYAD_LOGGER STREQUAL "CPP_LOGGER")
                        ${CPP_LOGGER_PATH} ${cpp-logger_PATH}
                        $ENV{CPP_LOGGER_DIR} $ENV{cpp-logger_DIR}
                        $ENV{CPP_LOGGER_PATH} $ENV{cpp-logger_PATH})
-    if (${CPP_LOGGER_FOUND})
+    if (${cpp-logger_FOUND})
```
Comment (hariharan-devarajan) [inline CMakeLists.txt:234]: You can look for lib version 1.0.0 in the find package. Also, switch CI to install 0.0.4. Thoughts?

**Review by hariharan-devarajan — CHANGES_REQUESTED** [review]
Few small comments.

File: pydyad/setup.cfg (hunk 1 above)
Comment (ilumsden) [reply pydyad/setup.cfg:12]: I didn't realize that you had pinned a version. I'll switch to that.

File: CMakeLists.txt (hunk 2 above)
Comment (ilumsden) [reply CMakeLists.txt:234]: Not sure I really understand what you mean here. We can discuss briefly in our meeting later today.

Comment (hariharan-devarajan) [reply CMakeLists.txt:234]: I have a PR for spack to include latest cpp logger. https://github.com/spack/spack/pull/45033 Once this is approved we can do this change.

Comment (ilumsden) [reply CMakeLists.txt:234]: Ah, I see now. Yeah, we should hold off until that gets merged.

**Comment by hariharan-devarajan** [comment]
@ilumsden We will have to switch DYAD_PROFILER macros to use DFTracer as well.

**Comment by ilumsden** [comment]
Good point. I was only thinking about it from the Python side since that was the only place where things were failing in CI. But, if I'm going to make these changes in the Python API, I should do it in the main body of DYAD too. I'll work on that after lunch

**Comment by hariharan-devarajan** [comment]
I recommend doing a search on DLIO_PROFILER and replacing it with DFTRACER.

You will have macros, apis, env variables.

**Comment by ilumsden** [comment]
@hariharan-devarajan I made all the changes you requested.

**Review by hariharan-devarajan — APPROVED** [review]

File: src/dyad/core/dyad_ctx.c [hunk 3]
```diff
@@ -286,24 +289,25 @@ dyad_rc_t dyad_init (bool debug,
     }
 
     ctx->relative_to_managed_path = relative_to_managed_path;
-    DYAD_LOG_INFO (ctx, "DYAD_CORE INIT: relative_to_managed_path %s", \
+    DYAD_LOG_INFO (ctx,
```
Comment (JaeseungYeom) [inline src/dyad/core/dyad_ctx.c:292]: This will end up in a compilation error.

File: src/dyad/modules/dyad.c [hunk 4]
```diff
@@ -233,29 +233,31 @@ dyad_fetch_request_cb (flux_t *h, flux_msg_handler_t *w, const flux_msg_t *msg,
         dyad_release_flock (mod_ctx->ctx, fd, &shared_lock);
         close (fd);
     }
-    DYAD_LOG_DEBUG(mod_ctx->ctx, "Close RPC message stream with an ENODATA (%d) message", ENODATA);
+    DYAD_LOG_DEBUG (mod_ctx->ctx, "Close RPC message stream with an ENODATA (%d) message", ENODATA);
     if (flux_respond_error (h, msg, ENODATA, NULL) < 0) {
-        DYAD_LOG_ERROR (mod_ctx->ctx, "DYAD_MOD: %s: flux_respond_error with ENODATA failed\n", __func__ );
+        DYAD_LOG_ERROR (mod_ctx->ctx,
```
Comment (JaeseungYeom) [inline src/dyad/modules/dyad.c:238]: missing '\'s

File: src/dyad/modules/dyad.c [hunk 5]
```diff
@@ -268,66 +270,68 @@ static void show_help (void)
...
+        "    -e, --error_log: Specify the file into which to redirect\n"
+        "                     error logging. Does nothing if DYAD was\n"
+        "                     not configured with '-DDYAD_LOGGER=PRINTF'\n"
+        "                     Need a filename as an argument.\n");
 }
 
 struct opt_parse_out {
-    const char* prod_managed_path;
-    const char* dtl_mode;
+    const char *prod_managed_path;
```
Comment (JaeseungYeom) [inline src/dyad/modules/dyad.c:289]: Let's refrain from styling changes like these. Unless the pointers are exclusively used by 'C', stick to the convensional C++ style.

**Comment by JaeseungYeom** [comment]
There are many needless styling modifications in this PR. Refrain from that in the future and focus on what the PR intends to fix.

File: src/dyad/core/dyad_ctx.c (hunk 3 above)
Comment (hariharan-devarajan) [reply src/dyad/core/dyad_ctx.c]: I updated the formatting changes

File: src/dyad/modules/dyad.c [hunk 6]
```diff
@@ -185,27 +185,27 @@ dyad_fetch_request_cb (flux_t *h, flux_msg_handler_t *w, const flux_msg_t *msg,
...
+        inlen = read (fd, inbuf + sizeof (file_size), file_size);
 #else
         inlen = read (fd, inbuf, file_size);
 #endif
         if (inlen != file_size) {
-            DYAD_LOG_ERROR (mod_ctx->ctx, \
-                            "DYAD_MOD: Failed to load file \"%s\" only read %zd of %zd.", \
-                            fullpath, \
-                            inlen, \
+            DYAD_LOG_ERROR (mod_ctx->ctx,
```
[the remark this answers is not in the document]
Comment (hariharan-devarajan) [reply src/dyad/modules/dyad.c]: I updated the formatting changes

File: src/dyad/modules/dyad.c (hunk 4 above)
Comment (hariharan-devarajan) [reply src/dyad/modules/dyad.c]: I updated the formatting changes

**Review by JaeseungYeom — APPROVED** [review]
With the fixes of missing backslashes, all looks good.

**Pull request merged**